app/objects/c_agent.py

```diff
@@ -290,30 +290,43 @@ async def all_facts(self):
     def executor_change_to_assign(self):
         return self._executor_change_to_assign
 
-    @executor_change_to_assign.setter
-    def executor_change_to_assign(self, executor_change_dict):
-        """Set pending executor change dict for the agent."""
-        if executor_change_dict:
-            self._executor_change_to_assign = executor_change_dict
-            if executor_change_dict.get('action') == 'remove':
+    def set_pending_executor_removal(self, executor_name):
+        """Mark specified executor to remove.
+
+        :param executor_name: name of executor for agent to remove
+        :type executor_name: str
+        """
+        if executor_name and isinstance(executor_name, str):
+            if executor_name in self.executors:
                 # Remove the executor server-side so planners can generate appropriate links immediately.
-                self._remove_executor(executor_change_dict.get('executor'))
+                self.executors.remove(executor_name)
+                self._executor_change_to_assign = dict(action='remove', executor=executor_name)
+
+    def set_pending_executor_path_update(self, executor_name, new_binary_path):
+        """Mark specified executor to update its binary path to the new path.
+
+        :param executor_name: name of executor for agent to update binary path
+        :type executor_name: str
+        :param new_binary_path: new binary path for executor to reference
+        :type new_binary_path: str
+        """
+        if executor_name and new_binary_path and isinstance(executor_name, str) and isinstance(new_binary_path, str):
```

Comment:
would it be beneficial to throw an else to these conditionals to print a debug message if someone configured things wrong so they dont have to step through the code?

Reply:
I can set that up. Would "error" be too strong?

Reply:
Ok, and maybe not, since most users may not pass in the `debug` logging flag.  The general idea is to make it easier for people just touching the ability files to see where they might have some issues in configuration.

Reply:
Makes sense. And these executor changes wouldn't typically come from ability files (we don't currently have that set up yet for abilities, though that could certainly be a future capability) - right now they're just set going to come from planners. At least that's the current use case I have.